filebeat/input/filestream/prospector_test.go

```diff
@@ -438,13 +504,18 @@ func (t *testHarvesterGroup) StopGroup() error {
 }
 
 type mockFileWatcher struct {
+	ctx         context.Context
 	nextIdx     int
 	events      []loginp.FSEvent
 	filesOnDisk map[string]os.FileInfo
 }
 
 func (m *mockFileWatcher) Event() loginp.FSEvent {
+AGAIN:
 	if len(m.events) == m.nextIdx {
+		if m.ctx != nil && m.ctx.Err() == nil {
+			goto AGAIN
```

Comment:
Can this busy waiting be replaced by `<-m.ctx.Done()`?  The only time `err != nil` is when `ctx.Done()` has been written to, from https://pkg.go.dev/context#Context:

```go
// If Done is not yet closed, Err returns nil.
// If Done is closed, Err returns a non-nil error explaining why:
// Canceled if the context was canceled
// or DeadlineExceeded if the context's deadline passed.
// After Err returns a non-nil error, successive calls to Err return the same error.
Err() [error](https://pkg.go.dev/builtin#error)
```

Reply:
Good point. It was rather a placeholder code, I forgot to get back to. Thanks.